item-impl/src/main/java/com/example/auction/item/impl/ItemServiceImpl.java

```diff
@@ -71,67 +74,73 @@ public ItemServiceImpl(PersistentEntityRegistry registry, BiddingService bidding
 
 
     @Override
-    public ServiceCall<Item, Item> createItem() {
+    public ServiceCall<ItemData, Item> createItem() {
         return authenticated(userId -> item -> {
-            if (!userId.equals(item.getCreator())) {
-                throw new Forbidden("User " + userId + " can't created an item on behalf of " + item.getCreator());
-            }
+            // ItemData doesn't contain a creatorId so the Principal _is_ the creator.
             UUID itemId = UUIDs.timeBased();
-            PItem pItem = new PItem(itemId, item.getCreator(), item.getTitle(), item.getDescription(),
-                    item.getCurrencyId(), item.getIncrement(), item.getReservePrice(), item.getAuctionDuration());
-            return entityRef(itemId).ask(new PItemCommand.CreateItem(pItem)).thenApply(done -> convertItem(pItem));
+            PItemData details = Mappers.fromApi(item);
+            // TODO: remove this PItem from here and pass the 3 values in the command?
+            PItem pItem = new PItem(itemId, userId, details);
+            return entityRef(itemId)
+                    .ask(new CreateItem(pItem))
+                    .thenApply(done -> Mappers.toApi(pItem));
         });
     }
 
     @Override
-    public ServiceCall<Item, Done> updateItem(UUID id) {
+    public ServiceCall<ItemData, Item> updateItem(UUID itemId) {
         return authenticated(userId -> item -> {
-            // todo implement
-            return null;
+            // userId may not match the item's creator Id. This is validated at Command Processing in the PersistentEntity
```

Comment:
If I want to review authorisation on the `ServiceImpl` I'll have to do an extra call to the `PersistentEntity`. The previous code was comparing the principal on the request with the `creatorId` in the body of the request which is not valid. I could cook a request with my credentials as principal over your Item setting on the body my creatorId.

I'm not sure this is particular example sample but I guess that demonstrating that `ServiceImpl` should avoid calling twice the same PersistentEntity as that'll lead to race conditions. I agree that since `creatorId` is never modified it'd be safe to do it in this case.

About 409/403... that's an error. When user is not allowed it should be mapped to a 403.